### PR TITLE
Move select2 import from baseframe to webpack

### PR DESCRIPTION
**Pull request by vidya-ram**
1. Move select2 import from baseframe to webpack
2. Change select2 to a dynamic import to reduce bundle size for pages that don't use select2 and improve page performance.

### CODE REVIEW

File: funnel/__init__.py [hunk 1]
```diff
@@ -135,7 +135,6 @@
     requires=['funnel'],
     ext_requires=[
         'pygments',
```
Comment (jace) [inline funnel/__init__.py:137]: We're not using Pygments anymore. That also can be removed.

File: funnel/assets/js/utils/form_widgets.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ export const Widgets = {
     };
     this.activateToggleSwitch(checkboxId, onSuccess);
   },
-  activate_select2() {
+  activateSelect2Focus() {
     /* Upgrade to jquery 3.6 select2 autofocus isn't working. This is to fix that problem.
       select2/select2#5993  */
```
Comment (jace) [inline funnel/assets/js/utils/form_widgets.js:65]: This ticket is marked as closed as jQuery 3.7 fixes it. Do we still need this patch?

**Review by jace — APPROVED** [review]
Two side issues, not directly affecting this PR.

File: funnel/__init__.py (hunk 1 above)
Comment (vidya-ram) [reply funnel/__init__.py:137]: Sure. Will take it up as a new PR

File: funnel/assets/js/utils/form_widgets.js (hunk 2 above)
Comment (vidya-ram) [reply funnel/assets/js/utils/form_widgets.js:65]: Will open a new PR to upgrade to jQuery 3.7 and remove this fix

**Pull request merged**